autogen/main/cluster.tf.tmpl

```diff
@@ -113,17 +113,15 @@ resource "google_container_cluster" "primary" {
   monitoring_service = local.logmon_config_is_set ? null : var.monitoring_service
   {% endif %}
   dynamic "monitoring_config" {
-    for_each = length(var.monitoring_enabled_components) > 0 || var.monitoring_enable_managed_prometheus ? [1] : []
-
-    content {
-      enable_components = length(var.monitoring_enabled_components) > 0 ? var.monitoring_enabled_components : []
-
-      dynamic "managed_prometheus" {
-        for_each = var.monitoring_enable_managed_prometheus ? [1] : []
-
-        content {
-          enabled = var.monitoring_enable_managed_prometheus
-        }
+    {% if beta_cluster %}
+    for_each = local.cluster_telemetry_type_is_set || local.logmon_config_is_set ? [1] : []
+    {% else %}
+    for_each = local.logmon_config_is_set || local.logmon_config_is_set ? [1] : []
+    {% endif %}
+    content{
+      enable_components = var.monitoring_enabled_components
+      managed_prometheus {
```

Comment:
Thanks @ericyz - As this is change from a dynamic to static block, likely some existing clusters will see config churn/change?  As as the gating variables are different.  Should this be a breaking change?

Reply:
Good point. It should be a break change

Reply:
Good point. It should be a breaking change